File: RayVE.CSharp/Output/PPMColor.cs
using RayVE.Extensions;
using System;
using System.Collections.Generic;

namespace RayVE.Output
{
    public class PPMColor
    {
        private readonly Color _color;

        public PPMColor(Color color) => _color = color;

        public IEnumerable<string> ToPPM(int maxValue)
            => new List<string>()
            {
                $"{GetPPMValue(_color.R, maxValue)}",
                $"{GetPPMValue(_color.G, maxValue)}",
                $"{GetPPMValue(_color.B, maxValue)}"
            };

        private static int GetPPMValue(double rawValue, int maxValue)
            => Convert.ToInt32((rawValue * maxValue).Clamp(0.0d, maxValue));
    }
}

File: RayVE.FSharp/Output/PPMColor.fs
namespace RayVE

module PPMColor =
    let ppmValue raw max =
        Math.clamp(0, max, raw * max)

    let toPPM (color: Color) max =
        [|
            ppmValue color.R max |> string
            ppmValue color.G max |> string
            ppmValue color.B max |> string
        |]